plugins/include/amxmodx.inc

```diff
@@ -943,12 +943,32 @@ native is_user_authorized(index);
  *       1 to MaxClients range. That means you can safely use this native
  *       without manually verifying that the index is a valid client index.
  *
+ * @note Due back-compatibility return's true after bot proceed client_connect
+ *       stage (when bot's pdata isn't created yet).
+ *
+ * @note Use is_user_connectedex to check player against real connection state
+ * 
  * @param index     Client index
  *
  * @return          1 if client is connected, 0 otherwise
+ *                  always returns 1 if bot passed client_connect stage
  */
 native is_user_connected(index);
 
+/**
+ *
+ * Returns if the client is fully connected (proceed client_putinserver state).
+ *
+ * @note This does not throw an error if the provided index is out of the
+ *       1 to MaxClients range. That means you can safely use this native
+ *       without manually verifying that the index is a valid client index.
+ *
+ * @param index     Client index
+ *
+ * @return          1 if bot is connected, 0 otherwise
+ */
+native is_user_connectedex(const index);
```

Comment:
To suffix these unix-style names, "is_user_connected_ex" would be preferred. Though you might consider "is_client_connected" or "is_user_or_bot_connected" if you don't like suffixes.

Reply:
Just tell me how and i could rename them, im not against.